### PR TITLE
Fixing type attribute warning for transfer_action

### PR DESCRIPTION
**Pull request by sithhell**
This fixes warnings occuring after recent changes to cmake.

### CODE REVIEW

File: hpx/runtime/actions/basic_action.hpp [hunk 1]
```diff
@@ -861,6 +861,7 @@ namespace hpx { namespace serialization
 #define HPX_REGISTER_ACTION_1(action)                                         \
     HPX_REGISTER_ACTION_2(action, action)                                     \
 /**/
+#if defined(HPX_MSVC) || defined(HPX_MINGW)
```
Comment (msimberg) [inline hpx/runtime/actions/basic_action.hpp]: Why does this get rid of any warnings? It looks to me like the if and else branches are exactly the same...

Comment (sithhell) [reply hpx/runtime/actions/basic_action.hpp]: Good catch ... looks like I didn't commit the latest version...

**Comment by msimberg** [comment]
@sithhell this caused undefined references with GCC. Were the warnings only a problem with clang?

**Comment by sithhell** [comment]
No. The warnings were a problem with gcc, not with clang. Very strange... I
am using gcc here as well. I'll have a look.

**Pull request merged**